drivers/hmis_csv_importer/app/models/hmis_csv_importer/importer/importer.rb

```diff
@@ -829,20 +832,9 @@ def involved_project_ids
             klass.hud_key => hud_keys,
           ).find_each(batch_size: SELECT_BATCH_SIZE) do |source|
             destination = source.as_destination_record
-            destination.pending_date_deleted = nil
-            case klass.hud_key
-            when :PersonalID
-              destination.demographic_dirty = true
-            when :EnrollmentID
-              destination.processed_as = nil
-            when :ExitID
-              # These are tracked separately so we can bulk update them at the end
-              track_dirty_enrollment(destination.EnrollmentID)
-            end
-            # ensure we always have the right data_source_id
-            destination.data_source_id = data_source.id
+            prepare_destination_for_update(klass, destination)
             batch << destination
-            if batch.count == INSERT_BATCH_SIZE && !custom_augmentation?(klass)
+            if batch.count == INSERT_BATCH_SIZE
```

Comment:
I wonder if `prepare_destination_for_update` should just return the destination client to make it more clear what's going on, then this could be:
```suggestion
batch << prepare_destination_for_update(klass, destination)
            if batch.count == INSERT_BATCH_SIZE
```

Reply:
good thought, thanks